hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MulticastJoiner.java

```diff
@@ -118,7 +118,7 @@ public void searchForOtherClusters() {
                             logger.fine("Ignoring merge join response, since " + joinInfo.getAddress()
                                     + " is already a member.");
                         }
-                        return;
+                        continue;
```

Comment:
Continue processing `SplitBrainJoinMessage`s even if the last one received is ignored, so that all messages in the deque will be processed in one execution of the `SplitBrainHandler` as [originally intended](https://github.com/hazelcast/hazelcast/pull/9328#discussion_r89759399).